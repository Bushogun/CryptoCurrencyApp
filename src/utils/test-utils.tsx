import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

// Importa tipos definidos en tu tienda Redux
import type { AppStore, RootState } from '@/redux/store';

// Importa el reducer de tu slice de Redux
import cryptoReducer from '@/redux/features/crypto-slice';

// Define una interfaz extendida de RenderOptions para incluir initialState y store
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

// La función renderWithProviders renderiza el componente con el contexto de Redux proporcionado por el Provider
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Crea automáticamente una instancia de store si no se pasa uno
    store = configureStore({
      reducer: { currency: currencySlice },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // Wrapper es un componente que envuelve el componente a renderizar y provee el contexto de Redux a través del Provider
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Retorna un objeto que contiene el store y todas las funciones de consulta de RTL
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
