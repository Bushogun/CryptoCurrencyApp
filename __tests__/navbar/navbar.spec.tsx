import { Navbar } from "@/app/components/navbar/Navbar";
import { renderWithProviders } from "@/utils/test-utils";

describe("Navbar component", () => {
  jest.mock("next/router", () => ({
    useRouter: jest.fn(),
  }));

  it("should render correctly ", () => {
    const { container } = renderWithProviders(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
