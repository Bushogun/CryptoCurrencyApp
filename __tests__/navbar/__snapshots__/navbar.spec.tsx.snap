// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly  1`] = `
<div>
  <nav
    class="navbar"
  >
    <a
      href="/"
    >
      <img
        alt="Crab Space Invaders"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="45"
        src="/_next/image?url=%2Fcrab-image.png&w=128&q=75"
        srcset="/_next/image?url=%2Fcrab-image.png&w=64&q=75 1x, /_next/image?url=%2Fcrab-image.png&w=128&q=75 2x"
        style="color: transparent;"
        title="Logo FakeCompany"
        width="60"
      />
    </a>
    <ul>
      <li>
        <a
          href="/exchange-rate"
        >
          Rate
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
